### PR TITLE
Flesh out `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by cloudhead**
Add code guidelines and instructions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -3,20 +3,135 @@
...
+    let r = RefString::try_from(node.to_string()).unwrap();
+
+Use `expect` only if the function expects certain invariants that were not met,
+either due to bad inputs, or a problem with the environment; and include the
+expectation in the message. For example:
+
+    logger::init(log::Level::Debug)
+        .expect("logger must only be initialized once");
+
+### Module imports
```
Comment (FintanH) [inline CONTRIBUTING.md:60]: What about `mod` declarations?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -3,20 +3,135 @@
...
+    }
+
+Use the most descriptive names for globals:
+
+    pub const KEEP_ALIVE_DELTA: LocalDuration = LocalDuration::from_secs(30);
+
+### Function naming
+
+Stay concise. Use the function doc comment to describe
+what the function does, not the name. Keep in mind function are in the
```
Comment (FintanH) [inline CONTRIBUTING.md:97]:
```suggestion
what the function does, not the name. Keep in mind functions are in the
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (FintanH) [reply CONTRIBUTING.md:60]: Aye, if that's what we do then it'd be good to state it :)

**Comment by slack-coder** [comment]
What about a note re: checking issues + proposing changes there first?  Discussing the changes before putting the work in can save significant time and effort

**Comment by cloudhead** [comment]
Yeah good point although I'd like to save that for when we can actually point people to `rad issue` etc.

**Review by FintanH — APPROVED** [review]
LGTM

**Pull request merged**